lib/tasks/spellcheck.rake

```diff
@@ -0,0 +1,91 @@
+#
+# Rake task for checking spelling in the documentation files.
+# By default checks all *.md and *.html files.
+#
+# Supports custom dictionaries:
+#
+#  - global dictionary located in the tasks gem (lib/tasks/.spell.dict)
+#  - repository specific dictionary (.spell.dict in the root directory)
+#
+# The custom dictionaries contains one word per line.
+# The lines starting with '#' character are ignored (used for comments),
+#
+
+CUSTOM_DICTIONARY_FILE = ".spell.dict"
+
+# read the global and the repository custom dictionary
+def read_custom_words
+  # read the glodefault custom dictionary
+  dict_path = File.expand_path("../#{CUSTOM_DICTIONARY_FILE}", __FILE__)
+  custom_words = File.read(dict_path).split("\n")
+
+  if File.exist?(CUSTOM_DICTIONARY_FILE)
+    puts "Loading custom dictionary (#{CUSTOM_DICTIONARY_FILE})..." if verbose == true
+    # read the custom dictionary from the project directory
+    custom_words += File.read(CUSTOM_DICTIONARY_FILE).split("\n")
+  end
+
+  # remove comments
+  custom_words.reject! { |word| word.start_with?("#")}
+  # remove duplicates
+  custom_words.uniq!
+
+  custom_words  
+end
+
+namespace :check do
+  desc "Run spell checker (by default for *.md and *.html files in Git)"
+  task :spelling, :regexp do |t, args|
+
+    DEFAULT_SPELLCHECK_REGEXP = /\.(md|html)\z/
+
+    # raspell is an optional dependency, handle the missing case nicely
+    begin
+      require "raspell"
+    rescue LoadError
+      $stderr.puts "ERROR: Ruby gem \"raspell\" is not installed."
+      exit 1
+    end
+
+    # initialize aspell
+    speller = Aspell.new("en_US")
+    speller.suggestion_mode = Aspell::NORMAL
+    # ignore the HTML tags in the text
+    speller.set_option("mode", "html")
```

Comment:
The task is a bit long. I think factoring out the speller `require` and initialization would give just `speller = apell_speller` here.